Development/Porting.md

```diff
@@ -5,86 +5,220 @@ permalink: /Porting/
-
- ```text
-        Flash Region 1 (BIOS): 00500000 - 0001b000
-        Flash Region 2 (Intel ME): 00003000 - 0001afff
-  ```
-
-* But that means that ME will need to be neutered and flashed externally prior
-  of flashing the BIOS region and IFD over unlocked IFD.  Instructions for this
-  external flashing use case, freeing and modifying ifd to reduce ME size and
-  expend BIOS region in consequence of liberated space can be [found here](https://github.com/corna/me_cleaner/wiki/External-flashing#neutralize-and-shrink-intel-me-useful-only-for-coreboot)
-
-* Run `make BOARD=$BOARD_DIR` (where `$BOARD_DIR` is the board directory
-  (`x230`, `x230-flash` and others found under `./boards` directory) to setup
-  the coreboot tree, using the new coreboot config file.  This will create the
-  output directory `build/x86/$BOARD_DIR/*.rom`, the rom name should be named
-  `coreboot.rom`.
-
-* If things don't work, please open an issue on [https://github.com/osresearch/heads/issues](https://github.com/osresearch/heads/issues)
+./docker_repro.sh BOARD=new_board real.remove_canary_files-extract_patch_rebuild_what_changed
+```
+Then, inject any changes into the coreboot fork canary file so that the build system refetches, repatches, and rebuilds the coreboot fork:
+```shell
+echo "bogus" | sudo tee build/x86/coreboot-t480/.canary
+```
+Note, patches that attempt to create files that are not expected to exist but exist will fail, showing at console what files already existed that couldn't be created. In this case, you need to remove them manually `rm -rf`, and restart the build with `./docker_repro.sh BOARD=new_board` which will progress until each modules/* required per board config is successfully built.
+	
+Testing
+===
+
+For thorough testing—especially for a new board—using the following template may be beneficial.
+Tasks to check, Copy and paste this template into the GitHub message window when reporting test results. Mark completed tasks with an x inside the brackets [ ]. Replace yyy with the relevant information
+```
+- [ ] Successful external flash link to circleci: https://yyy from commit `yyy` using external programmer model `yyy` on `yyy` Voltage mode
+- [ ] Boots successfully after the flashing: 
+- [ ] Setting clock prompt on first reboot: ok if triggered correctly after initial flashing and cmos battery disconnected
+- [ ] Clean boot detected (no keyring, nothing installed on disk): usb boot proposed and followed: ok
+- [ ] Boots on usb: ok
+- [ ] OS `yyy` install and reboot: ok
+- [ ] Heads functionality- no pubkey detected, but OS detected -> OEM-Factory-reset proposed. Done with `yyy` hardwarekey e.g. nk3: ok
+- [ ] On reboot after re-ownership: generate new HOTP/TOTP: ok
+- [ ] On reboot: default boot proposes to choose boot default + TPM DUK: ok
+- [ ] TPM DUK boots OS version: ok
+- [ ] TPM DUK boots QuebesOS 4.2.4: ok
+- [ ] wifi works based on OS `yyy`: ok
+- [ ] PR0: ok. 
+* flashprog -p internal (not locked)
+* lock_chip (locks_
```

Comment:
lock_chip (locks the platform with PR0, if PR0 patch applied in fork or under patches/coreboot-X.Y.Z, applied, and coreboot config applying proper preparation of the platform so lock_chip can simply lock the platform thorough PR0

See prior work
- https://github.com/linuxboot/heads/pull/1373
- https://github.com/linuxboot/heads/pull/1818
- unmerged patchset upstream https://review.coreboot.org/c/coreboot/+/85278